contracts/stake-cw20-external-rewards/src/msg.rs

```diff
@@ -0,0 +1,55 @@
+use cosmwasm_std::Uint128;
+use cw20::{Cw20ReceiveMsg, Denom};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use stake_cw20::hooks::StakeChangedHookMsg;
+
+use crate::state::{Config, RewardConfig};
+pub use cw_controllers::ClaimsResponse;
+
+#[derive(Serialize, Deserialize, JsonSchema, Debug, Clone, PartialEq)]
+pub struct InstantiateMsg {
+    pub owner: Option<String>,
+    pub manager: Option<String>,
+    pub staking_contract: String,
+    pub reward_token: Denom,
+}
```

Comment:
If we're instantiating contracts like this from a frontend it would be really nice to be able to specify rewards duration here.

Reply:
done, good logic